File: 10193 All You Need Is Love/solution.c
#include <stdio.h>
#include <stdlib.h>

int GCD(long i, long j) {
    return j == 0 ? i : GCD(j, i % j);
}

int main() {
    int N;
    scanf("%d", &N);
    char S1[17], S2[17];
    for(int i = 1; i <= N; ++i) {
        scanf("%s %s", S1, S2);
        printf("Pair #%d: %s!\n", i, GCD(strtol(S1, NULL, 2), strtol(S2, NULL, 2)) > 1 ? "All you need is love" : "Love is not all you need");
    }
    return 0;
}

File: 10193 All You Need Is Love/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer st = new StreamTokenizer(br);
        st.nextToken();
        int N = (int)st.nval;
        StringBuilder output = new StringBuilder();
        for(int i = 1; i <= N; ++i)
            output.append("Pair #" + i + ": " + (GCD(Integer.parseInt(br.readLine(), 2), Integer.parseInt(br.readLine(), 2)) > 1 ? "All you need is love" : "Love is not all you need") + "!\n");
        System.out.print(output);
	}
	
	static int GCD(int i, int j) {
        return j == 0 ? i : GCD(j, i % j);
    }
}
